### DISCUSSION TITLE
To the Sanni & the rest of the community...

### DISCUSSION THREAD

**Discussion by Ancyker** [sanni/cartreader#831] [General Chat]
Hi everyone. So, I've got a bit of a sappy/sentimental post here, feel free to skip it if it's not something you care about as it has nothing to do with the development side of the project. If you care a little bit but don't want to read this entire book yet still want to know what this post is about, I put a "tl;dr" note toward the end of the post in bold+italics. You can skip to that and still know what this is all about.

I'm a software developer/engineer by profession. I started learning various programming/scripting languages way back when I was around 16, right around 2000-2001 (god, I'm old). It started off as a hobby, I loved computers, and making them do stuff was so much fun. But, I didn't really start out wanting to code and such for a living. 

I built my first PC in 2000ish; long before there were YouTube guides telling you how. I enjoyed it a lot and I liked working with hardware, so I wanted to get a job fixing PCs. Somewhat luckily, one of my teachers told me that, although it was somewhat in demand and profitable at the time, he didn't see it staying that way and suggested I focus more on the software side since I seemed to enjoy that too. I respected him as he was the most knowledgeable person I knew when it came to both PCs and electronics in general. He knew some coding and had formal training in EE, so, I did just that. Messing with hardware stayed a hobby, building and fixing my/family/friend's PCs, all the while I taught myself more about coding. 

I started off working on some FOSS, contributing what I could when I could. This was back before that acronym even existed, or at least before it was known to me, and before git or even subversion; the first project I worked on used CVS and we communicated via a bug tracker and IRC. Eventually, I started freelancing as a web developer, and then I moved into a full-time salaried position.

So, what's the point of telling you all this? What does it have to do with the OSCR? Well, remember, my first choice was hardware. Working on software was/is fun, but it wasn't *my* first choice. I've worked on various little electronics projects over the years, 99% of which were just using Arduinos to control LED strips. I never really advanced my love of hardware past that, every year or so I'd build a new PC and the occasional server, but that's where my physical hobby ended. 

It... sucked. Like, I quite literally had no regularly-doable hobby that involved doing something with my hands. I like driving fast/racing, but track time is expensive (and so are speeding tickets and gas) and it's fairly seasonal. I tried learning an instrument, didn't stick. Tried learning to make fireworks, and did learn, but got bored after 2-3 months of it (and it was seasonal as well). I tried a *ton* of hobbies, but none ever really stuck...

That is until I found my old games and realized the saves were still intact. The first solution I came across was the [GB Operator](https://www.epilogue.co/product/gb-operator) by Epilogue. It worked just fine for GB/GBC/GBA games and I still use it to this day, but I had other systems and other games, what about those? I started looking for something that could handle N64 cartridges but couldn't seem to find anything that was still being produced. 

Then, one fateful day not long after (literally like a week or two), Linus Tech Tips posted [a video](https://www.youtube.com/watch?v=jISrg3V9ubo) on the @SAVE-THE-HERO HW3 variant. I'm not sure if STH sent it to them/asked them to do it or if LTT reached out, but regardless, that's how I found out about the OSCR. It looked a little janky (sorry, lol, no offense intended) but when they mentioned it was open source, I had to look more into it.

_A quick aside, I've noticed a few people on here seem to have an issue with STH for a reason I either do not know or do not understand. So, I'd like to just point out that I likely wouldn't be here if they didn't do what they do and weren't featured on LTT. I'm sure there are others here that could say the same. But I digress._

I believe this was right around the time Sanni had just moved on to HW5. My first main PCB was a HW5 Rev1 or maybe a Rev2. The six-slot, which I still use (for now), was a Rev1 with the messed up CIC unlock traces -- even has the bodge wires. I ordered the parts sometime in June 2022 and built my first OSCR around the end of July/start of August 2022. It was so much fun and was a great learning experience with soldering. Before the OSCR my soldering experience was mostly soldering a few headers and a bunch of wires.

_**For a tl;dr, start reading here.**_

And now we get to the meat of this pseudo-book I've been writing/you've been reading: The OSCR gave me a hobby with a hands-on experience that I've been in desperate need of. As I said, aside from occasionally building a computer or server, I haven't had much in the way of physical hobbies until now. If it wasn't for Sanni, STH, LTT to some extent, and all of you, I wouldn't have this wonderful hobby and project to work on. 

Everyone here is so passionate about these retro games. Most people would and do just download them off of the Internet, but here we all are, going through all this effort to do it "the right way". So finally, to my point: I think you all are amazing, I genuinely love this project and am glad so many people have come together to work on it. Thank you all, sincerely. I'm doing what I can to make this project the best it can be, just as all of you do.

Also, if any of you contributors out there have a need for hosting of some sort for this or another similar/related project, I have a full rack in one of [Hurricane Electric](https://he.net/)'s data centers and would be happy to provide it (hosting) to you, no charge, no strings. Just hit me up in Discord.

Once again, thank you all. :)

**Comment by cmcm711** [comment]
That's awesome! I first heard about the OSCR when I got a bug up my ass to replace all the batteries in my retro game collection, finding out during the research phase that the saves disappear when the battery is removed. I searched for a way to keep the saves intact during the battery swap and was disappointed in most options available, except for an old forum post I came across by Sanni. He had created a device that could dump ROMs and SRAM from multiple game system cartridges and rewrite them as well, but sadly it wasn't for sale.

Having never even ordered a PCB before (at the time the GitHub project page wasn't even created yet, I think), I created an account on that forum and PMed Sanni to ask if he had any parts left for sale, as I was adept at soldering (used to mod PS2s back in the day) and could put them together myself. I honestly wasn't expecting a response, but he did so and let me know he had parts for only one unit of a prototype left that he would assemble and sell to me at cost plus shipping, which was like $70.

I received a prototype of what I now know was HW3, and it worked great at dumping the SRAM from all my SNES games and rewriting it after the battery replacement. I was over the moon and also felt as if I had something of a unique piece of hardware on my hands.

A couple years went by and I went to do more save backups and restores, and I had lost my set of instructions on how to use the device. I went looking for them and found the GitHub project page detailing the new hardware revisions. After reading through the assembly guides and how-tos over and over, I decided I wanted to try my hand at building a HW5 Rev1 myself. Ordered the recommended parts, soldered everything together, and it worked like a charm.

Ended up building a few more for friends, ordering my first 3D printer and learning how to use that so I could print my own frames, learning how to 3D model to remix and design my own frames, realizing there was a market for these, teaching my partner to solder so they could help build readers, and spurring our motivation to design other creative projects so they could quit their crappy retail job and work from home.

I love the retro hardware creator community and hope to eventually learn enough to give back on even a small level of what's been provided to all its fans. Thank you Sanni, Ancyker, and everyone else who has contributed to this amazing project.

**Reply by Ancyker** [reply]
After I put my first one together I just knew I wanted to help out the project. HW3 had the clock-select switch and HW5 does not, which makes HW5 feel more approachable. So, the first thing I wanted to do, and did, was to make using the OSCR less intimidating by removing the voltage switch. Most people know you can ruin stuff if you send the wrong voltage to it, so having a way to make that automatic, I felt, was a big thing that was needed. I also added the RTC because I noticed there was firmware support for it but no module was available.

Right now I'm looking to try to remedy #812 but I need an oscilloscope to actually see what's going on.
